Source/SwiftLintFramework/Rules/RuleConfigurations/FileHeaderConfiguration.swift

```diff
@@ -48,28 +38,99 @@ public struct FileHeaderConfiguration: RuleConfiguration, Equatable {
             throw ConfigurationError.unknownConfiguration
         }
 
+        // Cache the created regexes if possible.
+        // If the pattern contains the SWIFTLINT_CURRENT_FILENAME placeholder,
+        // the regex will be recompiled for each validated file.
         if let requiredString = configuration["required_string"] {
             self.requiredString = requiredString
-            requiredRegex = try NSRegularExpression(pattern: requiredString,
-                                                    options: [.ignoreMetacharacters])
+            if !requiredString.contains(FileHeaderConfiguration.fileNamePlaceholder) {
+                _requiredRegex = try NSRegularExpression(pattern: requiredString,
+                                                         options: FileHeaderConfiguration.stringRegexOptions)
+            }
         } else if let requiredPattern = configuration["required_pattern"] {
             self.requiredPattern = requiredPattern
-            requiredRegex = try .cached(pattern: requiredPattern)
+            if !requiredPattern.contains(FileHeaderConfiguration.fileNamePlaceholder) {
+                _requiredRegex = try .cached(pattern: requiredPattern)
+            }
         }
 
         if let forbiddenString = configuration["forbidden_string"] {
             self.forbiddenString = forbiddenString
-            forbiddenRegex = try NSRegularExpression(pattern: forbiddenString,
-                                                     options: [.ignoreMetacharacters])
+            if !forbiddenString.contains(FileHeaderConfiguration.fileNamePlaceholder) {
+                _forbiddenRegex = try NSRegularExpression(pattern: forbiddenString,
+                                                          options: FileHeaderConfiguration.stringRegexOptions)
+            }
         } else if let forbiddenPattern = configuration["forbidden_pattern"] {
             self.forbiddenPattern = forbiddenPattern
-            forbiddenRegex = try .cached(pattern: forbiddenPattern)
+            if !forbiddenPattern.contains(FileHeaderConfiguration.fileNamePlaceholder) {
+                _forbiddenRegex = try .cached(pattern: forbiddenPattern)
+            }
         }
 
         if let severityString = configuration["severity"] {
             try severityConfiguration.apply(configuration: severityString)
         }
     }
+
+    private func makeRegex(for file: File, using pattern: String,
+                           options: NSRegularExpression.Options, escapeFileName: Bool) -> NSRegularExpression? {
+        // Recompile the regex for this file
+        guard let fileName = file.path?.bridge().lastPathComponent else {
+            queuedFatalError("Expected to validate a file.")
```

Comment:
Should we just return regex without replacing the pattern in this case? I think it'd be better to just fail this rule than to exit the whole program

Reply:
Sure. I wasn't sure what to do here...